### DISCUSSION TITLE
MsSqlHook timeout while executing a stored procedure

### DISCUSSION THREAD

**Discussion by srudrani** [apache/airflow#24695] [General]
Hi,

I am using following MsSqlOperator with mssql default configurations to execute a stored procedure. Stored procedure simply purge the records from table based on some condition. My dag contains the following statement. This dag execute fine with lower values of the parameters- maxLoopAllowed, batchSize, but when the values are increased after a threshold e.g. maxLoopAllowed =400, batchSize=5000 thee number of records purged is less than expected. The SP is independently tested with above parameter values and it works fine.

`   sql_command = """ 
    EXECUTE [mySchema].[myTable] ?, ?, ?
    """
task = MsSqlOperator(task_id='run_transaction_details_purge',
                        mssql_conn_id='myConnectionId',
                        sql=sql_command,
                        parameters=(deleteBeforeDate, maxLoopAllowed, batchSize),
                        autocommit=True)`

`from airflow.models import BaseOperator
from airflow.utils.decorators import apply_defaults
from hooks.mssql_hook import MsSqlHook
class MsSqlOperator(BaseOperator):
    template_fields = ('sql',)
    template_ext = ('.sql',)
    ui_color = '#ededed'

    @apply_defaults
    def __init__(
            self, sql, mssql_conn_id='mssql_default', parameters=None,
            autocommit=False, database=None, *args, **kwargs):
        super(MsSqlOperator, self).__init__(*args, **kwargs)
        self.mssql_conn_id = mssql_conn_id
        self.sql = sql
        self.parameters = parameters
        self.autocommit = autocommit
        self.database = database

    def execute(self, context):
        self.log.info('Executing: %s', self.sql)
        hook = MsSqlHook(mssql_conn_id=self.mssql_conn_id,
                         schema=self.database)
        hook.run(self.sql, autocommit=self.autocommit,
                 parameters=self.parameters)`

**Comment by potiuk** [comment]
You have to check what really times it out. this is likely somethign in your deployment - most likely firwall that breaks the connection - not that (you have forgotten to mention how your deployment is done) some cloud providers have such firewalls and timeout closing long running idle connections - but you have to check with your provider/deplyment. Nothing Airlfow can have infuence on.
